docker-app/qfieldcloud/core/middleware/requests.py

```diff
@@ -1,17 +1,35 @@
+import io
+import shutil
+
+from constance import config
+
+
 def attach_keys(get_response):
     """
     QF-2540
-    Annotate request with a str representation of relevant fields, so as to obtain a diff
-    by comparing with the post-serialized request later in the callstack.
+    Annotate request with:
+    - a `str` representation of relevant fields, so as to obtain a diff by comparing with the post-serialized request later in the callstack;
+    - a byte-for-byte, non stealing copy of the raw body to inspect multipart boundaries.
     """
 
     def middleware(request):
+        # only report raw body of POST requests with a < 10MBs content length
+        if (
+            request.method == "POST"
+            and "Content-Length" in request.headers
+            and int(request.headers["Content-Length"])
+            < config.SENTRY_REQUEST_MAX_SIZE_TO_SEND
+        ):
+            input_stream = io.BytesIO(request.body)
+            output_stream = io.BytesIO()
+            shutil.copyfileobj(input_stream, output_stream)
+            request.body_stream = output_stream
+
```

Comment:
No idea whether this is possible:
But if sentry_sdk could be made available already here, couldn't we somehow check for some conditions and report raw_body already here:

```
if "file" not in request.data or not request.FILES.getlist("file"):
    sentry_sdk.capture_message(output_stream)
```
?

Reply:
We have not parsed the body yet, so we don't have `request.data`.